rcl_action/test/rcl_action/test_action_communication.cpp

```diff
@@ -57,6 +57,51 @@ class CLASSNAME (TestActionCommunication, RMW_IMPLEMENTATION) : public ::testing
     ret = rcl_action_client_init(
       &this->action_client, &this->node, ts, action_name, &client_options);
     ASSERT_EQ(ret, RCL_RET_OK) << rcl_get_error_string().str;
+
+    this->wait_set_server = rcl_get_zero_initialized_wait_set();
+    ret = rcl_action_server_wait_set_get_num_entities(
+      &this->action_server,
+      &this->num_subscriptions_server,
+      &this->num_guard_conditions_server,
+      &this->num_timers_server,
+      &this->num_clients_server,
+      &this->num_services_server);
+    ASSERT_EQ(ret, RCL_RET_OK) << rcl_get_error_string().str;
+    ret = rcl_wait_set_init(
+      &this->wait_set_server,
+      this->num_subscriptions_server,
+      this->num_guard_conditions_server,
+      this->num_timers_server,
+      this->num_clients_server,
+      this->num_services_server,
+      rcl_get_default_allocator());
+    ASSERT_EQ(ret, RCL_RET_OK) << rcl_get_error_string().str;
+    ret = rcl_action_wait_set_add_action_server(&this->wait_set_server, &this->action_server, NULL);
+    ASSERT_EQ(ret, RCL_RET_OK) << rcl_get_error_string().str;
+    rcl_reset_error();
+
+    this->wait_set_client = rcl_get_zero_initialized_wait_set();
+    ret = rcl_action_client_wait_set_get_num_entities(
+      &this->action_client,
+      &this->num_subscriptions_client,
+      &this->num_guard_conditions_client,
+      &this->num_timers_client,
+      &this->num_clients_client,
+      &this->num_services_client);
+    ASSERT_EQ(ret, RCL_RET_OK) << rcl_get_error_string().str;
+    ret = rcl_wait_set_init(
```

Comment:
Alternatively, the action wait set functions should support multiple clients/servers per wait set. i.e. we can put the action client and server into a single wait set. I'm fine leaving these tests using two wait sets though, I'll leave it up to you.

Reply:
done!